pandas/core/groupby/ops.py

```diff
@@ -500,7 +501,9 @@ def _disallow_invalid_ops(self, values: ArrayLike, how: str):
             This is either not a valid function for this dtype, or
             valid but not implemented in cython.
         """
-        dtype = values.dtype
+        if is_numeric:
+            # never an invalid op for those dtypes, so return early as fastpath
+            return
 
         if is_categorical_dtype(dtype) or is_sparse(dtype):
```

Comment:
if we're micro-optimizing, these can be replaced with isinstance(dtype, FooDtype) checks

Reply:
I was planning to do a follow-up PR with several other dtype check optimizations, so will include it there